### PR TITLE
Extend Qualities and Setup Default Groups

### PR DESCRIPTION
**Pull request by Qstick**
#### Database Migration
YES

#### Description
This PR adds many new qualities (mostly for identification in of users current files), additionally it adds default quality groups.

#### Todos
- [x] Tests

#### Issues Fixed or Closed by this PR
#65 
*

### CODE REVIEW

File: frontend/src/TrackFile/MediaInfo.js [hunk 1]
```diff
@@ -27,6 +28,16 @@ function MediaInfo(props) {
           !!audioChannels &&
           audioChannels.toFixed(1)
         }
+
+        {
+          !!audioCodec && !!audioChannels && !!audioBitRate &&
```
Comment (danielunderwood) [inline frontend/src/TrackFile/MediaInfo.js]: Probably just me not being familiar with js, but is this what you meant to have here?

File: src/NzbDrone.Core/Datastore/Migration/004_add_various_qualities_in_profile.cs [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+            Execute.WithConnection(ConvertProfile);
+        }
+
+        private void ConvertProfile(IDbConnection conn, IDbTransaction tran)
+        {
+            var updater = new ProfileUpdater3(conn, tran);
+
+            updater.AddQuality(13);
+            updater.MoveQuality(5, 0);
+            updater.CreateNewGroup(0, 1000, "Trash Quality Lossy", new[] { 24, 25, 26, 27, 28, 29, 30, 31, 32 });
```
Comment (danielunderwood) [inline src/NzbDrone.Core/Datastore/Migration/004_add_various_qualities_in_profile.cs]: Should these numbers be given as constants somewhere? They seem to be kind of random numbers at first glance.

Comment (Qstick) [reply src/NzbDrone.Core/Datastore/Migration/004_add_various_qualities_in_profile.cs]: They are quality ids defined in quality.cs

File: frontend/src/TrackFile/MediaInfo.js (hunk 1 above)
Comment (Qstick) [reply frontend/src/TrackFile/MediaInfo.js]: Yea, just shorthand null checks and conditionals

Comment (dabates) [reply frontend/src/TrackFile/MediaInfo.js]: Could you express this with the shorthand as follows?  I didn't run a test to see if it works this way in JS 
```javascript
(!!(audioCodec && audioChannels && audioBitrate))
```
I think I've used this syntax in some ES6 and TypeScript, not sure if core ECMA supports it

File: src/NzbDrone.Core/Datastore/Migration/004_add_various_qualities_in_profile.cs (hunk 2 above)
Comment (dabates) [reply src/NzbDrone.Core/Datastore/Migration/004_add_various_qualities_in_profile.cs]: I agree with Daniel.. shouldn't we be using ENUMS or Constants to more fluent?

**Comment by dabates** [comment]
This is just a style thing for me.. because of where I work.. 

But why use `var` instead of strongly typing things.  Especially on returns where you KNOW what the result will be?

Just a question, not a critique.

**Comment by Qstick** [comment]
It’s mostly just following along with current code style from upstream and because it’s easy for local variables, especially when it’s clear what the type is. We can change over if we think it helps readability.

File: src/NzbDrone.Core/Datastore/Migration/004_add_various_qualities_in_profile.cs (hunk 2 above)
Comment (Qstick) [reply src/NzbDrone.Core/Datastore/Migration/004_add_various_qualities_in_profile.cs]: Will update

File: frontend/src/TrackFile/MediaInfo.js (hunk 1 above)
Comment (Qstick) [reply frontend/src/TrackFile/MediaInfo.js]: Looking at this closer, I think we really want:
`((!!audioCodec && !!audioBitRate) || (!!audioChannels && !!audioBitRate)) &&`

We want to render the separator if audioBitRate and one of the other two properties exists, not just when all 3 exist.

**Pull request merged**